src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningServiceSaveDelegateTest.java

```diff
@@ -67,49 +67,59 @@ public void testUpdateWhenHashAndAccessionCoincide() throws AccessionCouldNotBeG
         SaveResponse<String> generatedAccessions = delegate.doSaveAccessions(getAccessioningService()
                 .getAccessionGenerator().generateAccessions(accessions));
 
-        assertEquals(3, generatedAccessions.getSavedAccessions().size());
+        assertEquals(2, generatedAccessions.getSavedAccessions().size());
+        assertEquals(1, generatedAccessions.getSaveFailedAccessions().size());
 
         TestTransaction.start();
         TestTransaction.flagForCommit();
-        repository.delete("id-test-3");
+        repository.delete("3323D8A8F66A0602CC59372E866DD8E116DCCDB2");
+        repository.delete("3323D8A8F66A0602CC59372E866DD8E116DCCDB3");
+        repository.delete("3323D8A8F66A0602CC59372E866DD8E116DCCDB4");
```

Comment:
No asserts after this?

Reply:
It's cleaning the house for other tests. @DataJpa annotation doesn't clean when you force it to commit transactions so that it checks the constraints correctly.